driver/src/price_estimation/kraken/api.rs

```diff
@@ -0,0 +1,272 @@
+use anyhow::{anyhow, Context, Result};
+use isahc::prelude::*;
+use serde::Deserialize;
+use serde_with::rust::display_fromstr;
+use std::collections::HashMap;
+use std::time::Duration;
+
+/// A trait representing a Kraken API client.
+///
+/// Note that this is not the full API, only the subset required for the
+/// retrieving price estimates for the solver.
+#[cfg_attr(test, mockall::automock)]
+pub trait KrakenApi {
+    /// Retrieves the list of supported assets.
+    fn assets(&self) -> Result<HashMap<String, Asset>>;
+    /// Retrieves the list of supported asset pairs.
+    fn asset_pairs(&self) -> Result<HashMap<String, AssetPair>>;
+    /// Retrieves ticker information (with recent prices) for the given asset
+    /// pair identifiers.
+    fn ticker<'a>(&'a self, pairs: &'a [&'a str]) -> Result<HashMap<String, TickerInfo>>;
+}
+
+/// An HTTP Kraken API Client.
+#[derive(Debug)]
+pub struct KrakenHttpApi {
+    /// The base URL for the API calls.
+    base_url: String,
+    /// An HTTP client for all of the HTTP requests.
+    client: HttpClient,
+}
+
+/// The default Kraken API base URL.
+pub const DEFAULT_API_BASE_URL: &str = "https://api.kraken.com/0/public";
+
+/// The default timeout used for HTTP requests to Kraken.
+pub const DEFAULT_TIMEOUT: Duration = Duration::from_secs(10);
+
+impl KrakenHttpApi {
+    pub fn new() -> Result<Self> {
+        KrakenHttpApi::with_url(DEFAULT_API_BASE_URL, DEFAULT_TIMEOUT)
+    }
+
+    pub fn with_url(base_url: &str, timeout: Duration) -> Result<Self> {
+        let client = HttpClient::builder()
```

Comment:
Should we be using the same client as the GasEstimation/RPC transport and hide all isahc specific implementation details?

It would also be useful to measure the round trip times/sizes for all requests we make and report them into prometheus.